src/quo2/components/graph/wallet_graph/utils.cljs

```diff
@@ -0,0 +1,29 @@
+(ns quo2.components.graph.wallet-graph.utils)
+
+(defn find-highest-value
+  [maps]
+  (reduce (fn [max-value map]
+            (let [value (:value map)]
+              (if (nil? max-value) ; Check if it's the first element
+                value
+                (if (> value max-value)
+                  value
+                  max-value))))
+          nil
+          maps))
+
+(defn downsample-data
+  [data max-array-size]
+  (let [data-size            (count data)
+        aggregation-interval (max (/ data-size max-array-size) 1)]
```

Comment:
It looks like we are cutting out the most recent portion of data if it doesn't fit into `max-array-size`, e.g. if we have `data=[1 2 3 4 5]` and `max-array-size=2` then only `[1 2 3]` are going to be used. Not a big deal on a big set, but we still would rather want to cut the head of that set in case if the set is ordered from the older value to newer (which is likely).

Reply:
I've refactored and simplified the function and added some unit test so expected outputs are more understandable. Basically it cuts the array into by X amount of steps to fit given max amount of elements (if it exceeds) and avoid rendering lots of unnecessary points which won't even be visible in mobile screens.

If I did not misunderstood your suggestion, I think in that case we could split the data array and pre-process it, and keep this function as simple as possible. Let me know if that makes sense.

Reply:
Do we know what are expected numbers for `(count data)` and `max-array-size`? I do like that a new implementation is shorter but it might have introduced another issue depending on data we are going to show. 

For instance, in the previous implementation we would cut the last portion of data if it wouldn't have enough elements in it, e.g. for `data=[1 2 3 4 5]` and `max-array-size=2` if would be split into `[1 2 3]` and `[3 4]` and then `[3 4]` would not be shown, which in some cases might mean we do not show the recent data, and that might not be optimal (specifically if the user wants to see that recent data and the dataset is relatively small).

In a new implementation by doing decimation we can get a noticeably lower number of elements in the resulting data set although we potentially have enough information to show more details of it. e.g. if we have `(count data) = 21` and `max-array-size=20` the result will have only 11 elements. That's not going to be a problem if that dataset is big enough (comparing to  `max-array-size`) or if `(count data)` is multiple of `max-array-size`.

That's why I'm asking whether we know what to expect here :)

Reply:
For clarification, this function is only meant to use with big data sets (> 500 points). As far as I've tested, 500 points at max works fine and is performant. (count data) could be any value, maybe price history of a token, but for the preview I am using random datasets of 182500 elements (which would not be performant to render all and there's no space in the mobile screen to render them either way, so it is downsampled to 500 elements). In reality, I think user could view other smaller timeframes to inspect detailed information.
Anyways, I will create an issue so we can revisit this later once we start testing it with real data sets.

Reply:
Created it here https://github.com/status-im/status-mobile/issues/16831